### PR TITLE
axi_core: axi_adc_core: add code doc

### PR DESCRIPTION
**Pull request by amiclaus**
Add documentation for the axi_adc structures and functions.

No functional changes.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
v2:
- remove extra empty line

File: drivers/axi_core/axi_adc_core/axi_adc_core.h [hunk 1]
```diff
@@ -117,17 +117,35 @@
...
 	uint32_t base;
+	/** Number of channels */
 	uint8_t	num_channels;
+	/** AXI ADC Clock */
 	uint64_t clock_hz;
+	/** AXI ADC Channel Mask*/
 	uint32_t mask;
 };
 
+
```
Comment (rbolboac) [inline drivers/axi_core/axi_adc_core/axi_adc_core.h:138]: line can be removed.

**Review by amiclaus — COMMENTED** [review]
v3:
- remove another empty line

**Review by rbolboac — APPROVED** [review]
lgtm

**Pull request merged**